### ISSUE TITLE
Full page screenshot?

### ISSUE THREAD

**Issue by youshy** [ysmood/rod#32]
Is there a way in which we could get a screenshot of the whole page full size, not a screenshot of the window size?

**Comment by ysmood** [comment]
How about to use `page.PDF()`?

**Comment by youshy** [comment]
I've ended up writing this:

```go
func UseRod(link string) {
	url := launcher.New().
		Headless(false).
		Devtools(false).
		Launch()

	browser := rod.New().
		ControlURL(url).
		Trace(false).
		Connect().
		Timeout(time.Minute)

	defer browser.Close()
	page := browser.Timeout(time.Minute).Page(link)
	wait := page.WaitRequestIdle()
	wait()
	height := page.Eval(`() => document.body.clientHeight`).String()
	heightToInt, err := strconv.ParseInt(height, 10, 64)
	if err != nil {
		log.Fatalln(err)
	}
	page.Viewport(1920, heightToInt, 1, false)
	toByte, err := page.ScreenshotE(&proto.PageCaptureScreenshot{})

	f, err := os.Create("img_2.png")
	if err != nil {
		log.Fatal(err)
	}
	defer f.Close()
	img, _, _ := image.Decode(bytes.NewReader(toByte))
	err = png.Encode(f, img)
	if err != nil {
		log.Fatal(err)
	}
}
```

but will try the page.PDF() as well!

**Comment by ysmood** [comment]
Let's keep this ticket open since the full-page screenshot is actually useful. I just don't have enough time to implement it. The common solution is to take multiple shots while scrolling down the page until it hit the bottom, then merge all those shots into one.

If you are interested in it, it great if you can help to implement it

**Comment by youshy** [comment]
Both `page.PDF()` and `page.PDFE()` returns

```
2020/05/19 15:37:54 {"code":-32000,"message":"PrintToPDF is not implemented","data":""}
```

@ysmood I'm more than happy with helping!

**Comment by ysmood** [comment]
If your chrome version is too old, it doesn't support the PDF API.

The way you code it is similar to puppeteer, which changes the size of the frame. But puppeteer uses `Emulation.setDeviceMetricsOverride` to do it:

https://github.com/puppeteer/puppeteer/blob/caaf4d2086036d4bf201587529dd3e3b38b6f9c6/src/Page.ts#L1058

I think this kind of solution is good to have.

**Comment by ysmood** [comment]
To make the chrome version consistent, you can download a specific version of chrome and specified the bin location:

`launcher.New().Bin("the_path_of_chrome")`

Or you can check this FAQ to use docker: https://github.com/ysmood/rod#q-how-to-use-rod-with-docker

**Comment by ysmood** [comment]
Published `v0.34.2` with the full page screenshot feature.

I changed the name to `SceenshotFullPage`, so that when user types `Screenshot` the IDE will give the user a list to choose which one to use.

Because of the retina scale bug, I have to revert the `GetViewport`, the safest way to do it is to store a private viewport whenever we set it, once chrome adds the API to retrieve the metrics we can optimize it.
